### DISCUSSION TITLE
How to access currently selected Select value as it changes

### DISCUSSION THREAD

**Discussion by chairmansir** [mantinedev/mantine#921] [Q&A] [answered]
I am trying to access the value of a Select or NativeSelect control as it changes, to update my interface but it seems the OnChange handler does not get triggered. I added a console to it to see if i can access the value but nothing gets logged. But the label and value of the control gets changed everytime something is selected.

`<NativeSelect
                                          placeholder='Pick one (searchable)'
                                          required
                                          value={department}
                                          onChange={(event) => {
                                            console.log('val', event);
                                            setDepartment(
                                              event.currentTarget.value
                                            );
                                          }}
                                          {...form.getInputProps('department')}
                                          data={[
                                            {
                                              value: 'Clinical Pathology',
                                              label: 'Clinical Pathology',
                                            },
                                            {
                                              value: 'Community Health',
                                              label: 'Community Health',
                                            },
                                          ]}
                                        />`

**Comment by rtivital** [comment, marked as answer]
See controlled example – https://mantine.dev/core/native-select/#controlled
